File: dotNet/CTDemo/App_Code/Metadata.cs
// <summary>
// Defines SCT and SCT-AU Metadata (constants) utilised by the application. 
// </summary>
namespace CTDemo
{
    /// <summary>
    /// Defines the possible attribute values (each being a concept SCTID) assignable to a member in the 
    /// Australia Dialect Reference Set (ADRS).  
    /// </summary>
    
    public class Metadata
    {
        /** 
         * SNOMED CT RF2 'active' value
         */
        public const string ACTIVE_STATUS_VALUE = "1";

        public enum LanguageAcceptability : long
        {
            NONE = 0,
            PREFERRED = 900000000000548007L,
            ACCEPTABLE = 900000000000549004L
        }

        public static LanguageAcceptability GetLanguageAcceptability(long sctId){
            switch (sctId)
            {
                case (long)LanguageAcceptability.ACCEPTABLE:
                    return LanguageAcceptability.ACCEPTABLE;
                case (long)LanguageAcceptability.NONE:
                    return LanguageAcceptability.NONE;
                case (long)LanguageAcceptability.PREFERRED:
                    return LanguageAcceptability.PREFERRED;
                default:
                    return LanguageAcceptability.NONE;
            }
        }
    };

 }

File: java/src/main/au/gov/digitalhealth/model/Metadata.java
package au.gov.digitalhealth.model;

/**
 * Defines SCT and SCT-AU Metadata (constants) utilised by the application.  
 */
public class Metadata {

    /**
     * Defines the possible attribute values (each being a concept SCTID) assignable to a member in the 
     * Australia Dialect Reference Set (ADRS).  
     */
    public enum LanguageAcceptability {
        
        NONE(0), 
        PREFERRED(900000000000548007L), 
        ACCEPTABLE(900000000000549004L);
        
        private long sctId;
        
        private LanguageAcceptability(long sctId) {
            this.sctId = sctId;
        }
        
        public long getSctId() {
            return this.sctId;
        }
        
        public static LanguageAcceptability forSctId(Long sctId) {
            if (sctId == null) {
                return NONE;
            }
            for (LanguageAcceptability value : values()) {
                if (value.getSctId() == sctId) {
                    return value;
                }
            }
            return NONE;
        }
    };
    
    /** 
     * SNOMED CT RF2 'active' value
     */
    public final static String ACTIVE_STATUS_VALUE = "1";
}
